core/object/undo_redo.cpp

```diff
@@ -71,7 +71,13 @@ bool UndoRedo::_redo(bool p_execute) {
 	}
 
 	current_action++;
-	_process_operation_list(actions.write[current_action].do_ops.front(), p_execute);
+	
+	List<Operation>::Element *start_doops_element = actions.write[current_action].do_ops.front();
+	while (merge_total-- > 0 && start_doops_element) {
```

Comment:
We prefer not having decrement or increment in a statement like this, it hurts readability, please separate it

Reply:
Done